vpr/src/base/stats.cpp

```diff
@@ -129,6 +129,41 @@ void routing_stats(const Netlist<>& net_list,
     }
 }
 
+std::pair<vtr::NdMatrix<int, 3>, vtr::NdMatrix<int, 3>> calculate_channel_width() {
+    const auto& device_ctx = g_vpr_ctx.device();
+    const auto& rr_graph = device_ctx.rr_graph;
+
+    auto chanx_width = vtr::NdMatrix<int, 3>({{(size_t)device_ctx.grid.get_num_layers(),
+                                               device_ctx.grid.width(),
+                                               device_ctx.grid.height()}},
+                                             0);
+
+    auto chany_width = vtr::NdMatrix<int, 3>({{(size_t)device_ctx.grid.get_num_layers(),
```

Comment:
We might be able to use these to calculate more robust channel widths for drawing ... not sure if we already did that and not needed in this PR, but good to check to see.  Also could use for the linear congestion costs in the placer.
If you use more widely, we might want to move them to a new file under rr_graph/